test/unit/cloudantBaseService.test.js

```diff
@@ -166,4 +167,102 @@ describe('Test CloudantBaseService', () => {
     assert.ok(service.baseOptions.jar === undefined);
     assert.strictEqual(service.getAuthenticator().tokenManager.url, iamUrl);
   });
+
+  describe('Check keepAlive', () => {
+    const auth = new IamAuthenticator({
+      apikey: '1234',
+      clientId: '345',
+      clientSecret: '234',
+    });
+
+    const agentOptions = {
+      keepAlive: true,
+      maxSockets: 6,
+    };
+
+    function assertDefaultAgentOptions(service, agents) {
+      Object.keys(agentOptions).forEach((agentOption) => {
+        Object.values(agents).forEach((agent) => {
+          assert.strictEqual(
+            service.baseOptions[agent][agentOption],
+            agentOptions[agentOption]
+          );
+        });
+      });
+    }
+    describe('When it is set in the constructor', () => {
+      it('Default URL with https protocol should set both agents', () => {
+        BaseService.DEFAULT_SERVICE_URL = iamUrl; // iamUrl has https protocol
+        const service = new CloudantBaseService({
+          authenticator: auth,
+        });
+        assertDefaultAgentOptions(service, ['httpAgent', 'httpsAgent']);
+      });
+      it('No protocol should set both agents', () => {
+        const service = new CloudantBaseService({
+          authenticator: auth,
+          serviceUrl: newUrl, // newUrl has no protocol
+        });
+        assertDefaultAgentOptions(service, ['httpAgent', 'httpsAgent']);
+      });
+
+      it('Http protocol should set both agents', () => {
+        const service = new CloudantBaseService({
+          authenticator: auth,
+          serviceUrl: appleUrl, // appleUrl has http protocol
+        });
+        assertDefaultAgentOptions(service, ['httpAgent', 'httpsAgent']);
+      });
+
+      it('Custom keepAlive independent from protocol in URL', () => {
+        const service = new CloudantBaseService({
+          authenticator: auth,
+          serviceUrl: appleUrl, // appleUrl has http protocol
+          httpsAgent: new https.Agent({ keepAlive: false }),
+        });
+        assert.strictEqual(service.baseOptions.httpsAgent.keepAlive, false);
+        assertDefaultAgentOptions(service, ['httpAgent']);
+      });
+    });
+    describe('When it is set later', () => {
```

Comment:
```suggestion
    describe('Use customer defined Agent', () => {
```

Reply:
I removed the describe blocks and used better namings in the it blocks in 69652b5 and c55694c